File: daily_problems/2025/12/1213/personal_submission/cf105284c_Koishii.cpp
#include <bits/stdc++.h>
using namespace std;

const int mod = 1e9 + 7;
const int N = 1000006;
long long inv[N], h[N];

void init(){
	inv[1] = h[1] = 1;
	for (int i = 2; i < N; i++) {
		inv[i] = mod - mod / i * inv[mod % i] % mod;
		h[i] = (h[i - 1] + inv[i]) % mod;
	}
}

void soviet() {
	int n; std::cin >> n;
	std::cout << h[n] << '\n' << inv[n] << '\n';
	std::cout << (h[n] + inv[n] - 1) % mod << '\n';
}

int main() {
	int M_T = 1; std::ios::sync_with_stdio(false); std::cin.tie(nullptr);
	init();
	std::cin >> M_T;
	while (M_T--) { soviet(); }
	return 0;
}

File: daily_problems/2025/12/1213/personal_submission/cf105284c_Koishii.py
mod = 10 ** 9 + 7
N = 10 ** 6 + 6

inv = [1] * N
h = [1] * N

for i in range(2, N):
    inv[i] = mod - mod // i * inv[mod % i] % mod
    h[i] = (h[i - 1] + inv[i]) % mod

def soviet():
    n = int(input())
    print((h[n] + inv[n] - 1) % mod)


MT = int(input())
for _ in range(MT):
    soviet()
